Mage/src/main/java/mage/abilities/keyword/CraftAbility.java

```diff
@@ -0,0 +1,147 @@
+package mage.abilities.keyword;
+
+import mage.abilities.Ability;
+import mage.abilities.ActivatedAbilityImpl;
+import mage.abilities.costs.Cost;
+import mage.abilities.costs.CostImpl;
+import mage.abilities.costs.common.ExileSourceCost;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.Card;
+import mage.constants.Outcome;
+import mage.constants.TimingRule;
+import mage.constants.Zone;
+import mage.filter.FilterCard;
+import mage.filter.FilterPermanent;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetCardInGraveyardBattlefieldOrStack;
+import mage.util.CardUtil;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * @author TheElk801
+ */
+public class CraftAbility extends ActivatedAbilityImpl {
+
+    private final String description;
+    private final String manaString;
+
+    public CraftAbility(String manaString, String description, FilterCard filterCard, FilterPermanent filterPermanent) {
+        this(manaString, description, new TargetCardInGraveyardBattlefieldOrStack(1, 1, filterCard, filterPermanent));
```

Comment:
okay, so this target class defaults to a spell filter that matches nothing, that works fine but took some research to understand

Reply:
yeah it was something I added previously but I get how it's confusing